notebook2/notebook/notebook.py

```diff
@@ -389,6 +433,7 @@ def login_auth0():
 
 
 @app.route('/logout', methods=['POST'])
+@requires_auth()
```

Comment:
Awkward but seems appropriate.

Reply:
Why is authentication required? I assume they'll get redirected to auth0 who will give them some appropriate error message.

Reply:
I slightly preferred the meaning of /logout as the removal of an authenticated user's session. In that case, logging out when one isn't logged in is nonsensical. 

This also saves a trip to auth0 (which will redirect to the 'redirectTo' param I pass below this) and imo provides a neater API.

Reply:
Agreed it's nonsensical, but why force them to log in only to then be redirected to the logout page which will log them out? Seems more straightforward to send them to auth0 which, I hope, will produce an error.

Reply:
Sure, I'll get rid of it.